libsmack/libsmack.c

```diff
@@ -605,27 +610,29 @@ ssize_t smack_new_label_from_self(char **label)
 
 ssize_t smack_new_label_from_socket(int fd, char **label)
 {
-	char dummy;
+	char buf[SMACK_LABEL_LEN + 2];
```

Comment:
Describe the reason for having the additional byte in `buf` (why `+ 2` instead of `+ 1`).